lib/syskit/network_generation/engine.rb

```diff
@@ -311,16 +311,6 @@ def find_selected_device_in_hierarchy(argument_name, leaf_task, requirements)
                 _, model, _ = leaf_task.requirements.resolved_dependency_injection.selection_for(nil, requirements)
                 if model && dev = model.arguments[argument_name]
                     return dev
-                else
-                    devices = Set.new
```

Comment:
I see. Assuming we wanted to allow multiple devices per task. Let's think about options we have for auto-selection.

1) Automatically select the first device type that would match
2) Raise if there are multiple options

Other ideas? Do you have a preferred behavior?